src/proxy-ui-api/src/main/java/org/niis/xroad/securityserver/restapi/wsdl/UnsupportedOpenApiVersionException.java

```diff
@@ -0,0 +1,38 @@
+/*
+ *  The MIT License
+ *  Copyright (c) 2019- Nordic Institute for Interoperability Solutions (NIIS)
+ *  Copyright (c) 2018 Estonian Information System Authority (RIA),
+ *  Nordic Institute for Interoperability Solutions (NIIS), Population Register Centre (VRK)
+ *  Copyright (c) 2015-2017 Estonian Information System Authority (RIA), Population Register Centre (VRK)
+ *
+ *  Permission is hereby granted, free of charge, to any person obtaining a copy
+ *  of this software and associated documentation files (the "Software"), to deal
+ *  in the Software without restriction, including without limitation the rights
+ *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *  copies of the Software, and to permit persons to whom the Software is
+ *  furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included in
+ *  all copies or substantial portions of the Software.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ *  THE SOFTWARE.
+ */
+package org.niis.xroad.securityserver.restapi.wsdl;
+
+import org.niis.xroad.restapi.exceptions.ErrorDeviation;
+import org.niis.xroad.restapi.service.ServiceException;
+
+/**
+ * Thrown if the provided OpenAPI document version differs from 3.0.x
+ */
+public class UnsupportedOpenApiVersionException extends ServiceException {
```

Comment:
Perhaps this could be a subclass of the OpenApiParser.ParserException, that would avoid method signature and try-catch changes where handling that exception specially is not required; Or, maybe version mismatch could have been reported using the ParserException with specific ErrorDeviation without even introducing a new exception.

Reply:
True, but I think that the implementation is in line with the rest of the `proxy-ui-api` codebase. We have made an intentional decision to be quite verbose with the thrown exceptions. Of course, as a known downside, this might lead to some extra try-catch handling and also results in some of the method signatures being quite bloated with exceptions.

At the moment the parser is used only by `addServiceDescription`, `updateServiceDescription` and `refreshServiceDescription` and I think it's good to see the explicit exception in their signatures. Not sure how I feel about "hiding" the `UnsupportedOpenApiVersionException` under `ParsingException`.

Reply:
One could argue that the UnsupportedVersionException is a (subclass of) ParsingException, and at the controller level it is handled exactly the same as the parsing error -- just wrapped to a more general exception. Therefore polluting the signatures and forcing changes to higher level code seems unnecessary here. Anyway, I accept that this is in line with other code, and maybe we revisit the general issue some other time.